lib/AST/PlatformKind.cpp

```diff
@@ -187,20 +196,21 @@ PlatformKind swift::targetPlatform(const LangOptions &LangOpts) {
 
 bool swift::inheritsAvailabilityFromPlatform(PlatformKind Child,
                                              PlatformKind Parent) {
+  if (auto ChildPlatformBase = basePlatformForExtensionPlatform(Child)) {
+    if (Parent == ChildPlatformBase)
+      return true;
+  }
+
   if (Child == PlatformKind::macCatalyst && Parent == PlatformKind::iOS)
     return true;
 
   if (Child == PlatformKind::macCatalystApplicationExtension) {
     if (Parent == PlatformKind::iOS ||
-        Parent == PlatformKind::iOSApplicationExtension ||
-        Parent == PlatformKind::macCatalyst) {
```

Comment:
Just tracing the logic here, is this removal due to the above early exit on `if (Parent == ChildPlatformBase)`?

Reply:
Yeah, this check is now redundant with the new early exit earlier.